bin/packages/validate-typescript-version.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+const tscDetectedVersion = require( 'typescript' ).version;
+const tscDependencyVersion = require( '../../package.json' ).devDependencies
+	.typescript;
```

Comment:
Technically the latter of the two fits more the definition of an "internal dependency", as the dependency itself is the `package.json` fie.

Reply:
```suggestion
/**
 * External dependencies
 */
const tscDetectedVersion = require( 'typescript' ).version;

/**
 * Internal dependencies
 */
const tscDependencyVersion = require( '../../package.json' ).devDependencies
	.typescript;
```